staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -127,14 +148,47 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 		proxy = config.Proxy
 	}
 
-	transport := utilnet.SetTransportDefaults(&http.Transport{
+	transport := buildReloadableTransport(utilnet.SetTransportDefaults(&http.Transport{
 		Proxy:               proxy,
 		TLSHandshakeTimeout: 10 * time.Second,
 		TLSClientConfig:     tlsConfig,
 		MaxIdleConnsPerHost: idleConnsPerHost,
 		DialContext:         dial,
 		DisableCompression:  config.DisableCompression,
-	})
+	}))
+
+	if shouldReloadCA && tlsConfig != nil {
+		caFile := config.TLS.CAFile
+		caData := config.TLS.CAData
+		baseRT := transport.rt.Load().Clone()
+		go wait.UntilWithContext(wait.ContextForChannel(DialerStopCh), func(_ context.Context) {
+			data, err := os.ReadFile(caFile)
+			if err != nil {
+				klog.ErrorS(err, "failed to read CA file", "file", caFile)
+				return
+			}
+
+			if len(data) == 0 {
+				klog.InfoS("CA file is empty", "file", caFile)
+				return
+			}
+
+			if bytes.Equal(data, caData) {
+				return
+			}
+
+			rootCAs, err := rootCertPool(data)
+			if err != nil {
+				klog.ErrorS(err, "failed to build pool from CA file", "file", caFile)
+				return
+			}
+
+			newRT := baseRT.Clone()
+			newRT.TLSClientConfig.RootCAs = rootCAs
+
+			transport.rt.Store(newRT)
```

Comment:
how does this behave with existing connections?

Reply:
Connection tracking happens at the round tripper so once the old `rt` is unused those connections would be unused (presumably with some cleanup at the OS level once the Go GC cleans up).

Reply:
I think calling `CloseIdleConnections` on the old transport would be appropriate.

Reply:
I think you would need a more targeted approach to only close connections associated with the old transport (which is hard to do since the dialer is shared).

Reply:
> at the time of the reload because you have a new rootCA you'll only have connections that need to be disrupted

@aojea this is not true because the server will not close existing connections nor will the client need to verify the old connections with the new root (and servers are supposed to do graceful rotation of their roots anyway but even without that the old connections will work fine).

Reply:
oh, I see you want to do seamless rotation, I was assuming that you could do the same we do in kubelet, cleanup all the connections and restart with the new certificate all the connections ...

Reply:
We need the hard cut over with client certs because of how the API server handles authentication.  We don't need it for root CA changes, so it seems better to gracefully rotate.